README.md

```diff
@@ -54,35 +56,66 @@ cp .envrc.example .envrc
+If `direnv` hook is setup correctly, after making a change to the variables in `.envrc` file,
+`direnv` will request you to run `direnv allow` for the new set of variables to take effect.
 
 Also remember to update `.envrc.example` when removing, or introducing a new variable.
 
+### Start the development environment
 
-### Database setup
-
-First, make sure there is no other Postgres instance running locally, or using port 5432.
-
-Quickly start a Postgres database in docker with
+Once the database and environment variables are setup, run
 
 ```
-docker run --name naisten_linja_db -e POSTGRES_USER=nl_dev -e POSTGRES_PASSWORD=nl_dev -e POSTGRES_DB=nl_dev -v "/$(pwd)/db-data:/var/lib/postgresql/data" -p 5432:5432 postgres:11.9
+npm run dev
 ```
 
-Upon the first time, this will create a new container named `naisten_linja_db` with the credentials matching the default environment variables set in `.envrc` file.
+This will start a docker container running PostgreSQL, run database migration, then start both the front and backend
+development app.
 
-Run migrations:
+After that, the frontend app is accessible at http://localhost:3000 (`FRONTEND_PORT` was set in
+`.envrc` to 3000 by default).
 
-```
-npm run db-migrate up all
-```
+In case the database failed to start with this error `Bind for 0.0.0.0:5433 failed: port is already allocated`, it's
+most likely another service is using port `5433` in your local environment. Try stopping that service. Alternatively,
+change `DB_PORT` value in `.envrc` to a different one (`5434` for example), run `direnv allow && docker rm
+naisten_linja_db`. This will update `DB_PORT` and remove the old `naisten_linja_db` container that was using the old
+port forwarding setting. Then, try `npm run dev` again.
 
-### Start the development environment
+### Making changes to the database schema
 
-Once the database and environment variables are setup, run
+We are using [db-migrate](https://github.com/db-migrate/node-db-migrate) to manage database schema changes. All
+migration files are located in `./db/migrations/` folder, which should not be modified if they are already deployed to
+remote environments. New changes made to the database schema should be done through creating new migrations. Refer to
+db-migrate's [documentation page](https://db-migrate.readthedocs.io/en/latest/) for more details.
+
+A few basic `db-migrate` commands:
 
 ```
-npm run dev
+# Generate a new migration file at `./db/migrations/<timestamp>-your-migration-name.js`
+db-migrate create your-migration-name
+# Run all migrations
+db-migrate up
+# Run up to 3 migrations
+db-migrate up -c 3
+# Revert the latest migration
+db-migrate down
+# Revert up to 5 latest migrations
+db-migrate down -c 5
 ```
 
-This will start both the frontend and backend development environment. The development service should be accessible at http://localhost:3000
+### Modifications to `react-scripts` module
```

Comment:
Would be great to hear your input about this. If `patch-package` is too cringy, then we can try to solve the issue in another way? : D

Reply:
I think this is a well reasoned solution for now, let's see later then if we can change it if the future versions opens up simper solutions!